registration/registration-services/src/main/java/io/mosip/registration/dao/impl/UserOnboardDAOImpl.java

```diff
@@ -302,6 +309,27 @@ public String insertExtractedTemplates(List<BIR> templates) {
 				bioMetrics.setCrBy(SessionContext.userContext().getUserId());
 				bioMetrics.setCrDtime(Timestamp.valueOf(DateUtils.getUTCCurrentDateTime()));
 				bioMetrics.setIsActive(true);
+				
+				try (InputStream xsd = getClass().getClassLoader().getResourceAsStream("cbeff.xsd")) {
+					BiometricRecord biometricRecord = new BiometricRecord();
+					biometricRecord.setOthers(new HashMap<>());
+					biometricRecord.setSegments(Arrays.asList(template));
+					
+					CbeffContainerImpl cbeffContainer = new CbeffContainerImpl();
+		            BIR bir = cbeffContainer.createBIRType(biometricRecord.getSegments());
+		            HashMap<String, String> entries = new HashMap<String, String>();
+		            biometricRecord.getOthers().forEach((k, v) -> {
+		                entries.put(k, v);
+		            });
+		            bir.setOthers(entries);
+		            
+					bioMetrics.setBirData(CbeffValidator.createXMLBytes(bir, IOUtils.toByteArray(xsd)));
+					
+					bioMetrics.setBirMinorVersion(Integer.toString(template.getVersion().getMinor()));
+					bioMetrics.setBirMajorVersion(Integer.toString(template.getVersion().getMajor()));
+				} catch (Exception exception) {
+					LOGGER.error("Serialization of BIR failed with exception >> ", exception);
```

Comment:
We are simply ignoring the serialization failure?
Failure in serialization or save of biometrics should fail the operator onboarding.

Reply:
Modified exception handling to fail onboarding.